modules/objdetect/src/qrcode.cpp

```diff
@@ -3993,13 +4059,533 @@ bool QRCodeDetector::detectAndDecodeMulti(
     updatePointsResult(points_, points);
     decoded_info.clear();
     ok = decodeMulti(inarr, points, decoded_info, straight_qrcode);
-    updatePointsResult(points_, p->updateQrCorners);
+    updatePointsResult(points_, updateQrCorners);
     return ok;
 }
 
-void QRCodeDetector::setUseAlignmentMarkers(bool useAlignmentMarkers) {
-    p->useAlignmentMarkers = useAlignmentMarkers;
+QRCodeDetector& QRCodeDetector::setUseAlignmentMarkers(bool useAlignmentMarkers) {
+    (std::static_pointer_cast<ImplContour>)(p)->useAlignmentMarkers = useAlignmentMarkers;
+    return *this;
+}
+
+QRCodeDetectorAruco::Params::Params() {
+    minModuleSizeInPyramid = 4.f;
+    maxRotation = (float)CV_PI/12.f;
+    maxModuleSizeMismatch = 1.75f;
+    maxTimingPatternMismatch = 2.f;
+    maxPenalties = 0.4f;
+    maxColorsMismatch = 0.2f;
+    scaleTimingPatternScore = 0.9f;
+}
+
+struct FinderPatternInfo {
```

Comment:
BTW, it makes sense to hide local internal classes (especially with some generic names) into inner anonymous namespace.